orchagent/bfdorch.cpp

```diff
@@ -152,8 +155,52 @@ void BfdOrch::doTask(NotificationConsumer &consumer)
     }
 }
 
+bool BfdOrch::register_bfd_state_change_notification(void)
+{
+    sai_attribute_t  attr;
+    sai_status_t status;
+    sai_attr_capability_t capability;
+
+    status = sai_query_attribute_capability(gSwitchId, SAI_OBJECT_TYPE_SWITCH, 
+                                            SAI_SWITCH_ATTR_BFD_SESSION_STATE_CHANGE_NOTIFY,
+                                            &capability);
+
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Unable to query the BFD change notification capability");
+        return false;
+    }
+
+    if (!capability.set_implemented)
+    {
+        SWSS_LOG_ERROR("BFD register change notification not supported");
```

Comment:
why it is an error and not just warning? it is OK that some vendors does not support this option and BFD can work

Reply:
This error would be thrown when user explicitly configures BFD. In that case we need error LOG. It will not appear if BFD is not configured by the user

Reply:
I don't think deregistration is required. The only problem we might end up is registering callback when it is not supported and this PR handles it gracefully.
We have multiple other callbacks like port,fdb which are registered and there is no deregistering sequence.

Reply:
Agree with @dgsudharsan